File: import_schedule.ts
import fs from "fs";
import path from "path";
import { parse } from "csv-parse";
import mysql from "mysql2/promise";
import dotenv from "dotenv";

dotenv.config();

const CSV_FILE = path.join(__dirname, "Psj Spring 2026 Master schedule - Sheet1.csv");

const config = {
  host:     process.env.DB_HOST ?? "localhost",
  port:     parseInt(process.env.DB_PORT ?? "3306"),
  user:     process.env.DB_USER,
  password: process.env.DB_PWD ?? "",
  database: process.env.DB_NAME ?? "soccer_schedule",
};

if (!config.user) {
  console.error("DB_USER is not set in .env");
  process.exit(1);
}

const CREATE_TABLE = `
  CREATE TABLE IF NOT EXISTS games (
    id        INT AUTO_INCREMENT PRIMARY KEY,
    game_date DATE NOT NULL,
    game_time VARCHAR(10),
    location  VARCHAR(50),
    age       VARCHAR(10),
    gender    VARCHAR(10),
    division  VARCHAR(100),
    home_team VARCHAR(150),
    away_team VARCHAR(150)
  )
`;

const INSERT_ROW = `
  INSERT INTO games (game_date, game_time, location, age, gender, division, home_team, away_team)
  VALUES (?, ?, ?, ?, ?, ?, ?, ?)
`;

function parseDate(raw: string): string {
  const [m, d, y] = raw.trim().split("/");
  return `${y}-${m.padStart(2, "0")}-${d.padStart(2, "0")}`;
}

function parseTime(raw: string): string {
  // Strip timezone suffix (EST, EDT)
  return raw.trim().split(" ")[0];
}

async function main() {
  const conn = await mysql.createConnection(config);
  await conn.execute(CREATE_TABLE);

  const records: string[][] = await new Promise((resolve, reject) => {
    const rows: string[][] = [];
    fs.createReadStream(CSV_FILE)
      .pipe(parse({ relaxColumnCount: true, skipEmptyLines: false }))
      .on("data", (row: string[]) => rows.push(row))
      .on("end", () => resolve(rows))
      .on("error", reject);
  });

  // Skip header row
  const dataRows = records.slice(1);

  let inserted = 0;
  let skipped = 0;

  for (const row of dataRows) {
    // Skip blank separator rows
    if (!row[0]?.trim()) continue;

    try {
      const values = [
        parseDate(row[0]),
        parseTime(row[1]),
        row[2]?.trim() || null,
        row[3]?.trim() || null,
        row[4]?.trim() || null,
        row[5]?.trim() || null,
        row[6]?.trim() || null,
        row[7]?.trim() || null,
      ];
      await conn.execute(INSERT_ROW, values);
      inserted++;
    } catch (err) {
      console.warn(`Skipping row: ${err} — ${row}`);
      skipped++;
    }
  }

  await conn.end();
  console.log(`Done: ${inserted} rows inserted, ${skipped} skipped.`);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: import_schedule.py
import csv
import sys
import argparse
from datetime import datetime

try:
    import mysql.connector
except ImportError:
    print("mysql-connector-python not found. Install with: pip install mysql-connector-python")
    sys.exit(1)

CSV_FILE = "Psj Spring 2026 Master schedule - Sheet1.csv"

CREATE_TABLE = """
CREATE TABLE IF NOT EXISTS games (
    id          INT AUTO_INCREMENT PRIMARY KEY,
    game_date   DATE NOT NULL,
    game_time   VARCHAR(10),
    location    VARCHAR(50),
    age         VARCHAR(10),
    gender      VARCHAR(10),
    division    VARCHAR(100),
    home_team   VARCHAR(150),
    away_team   VARCHAR(150)
)
"""

INSERT_ROW = """
INSERT INTO games (game_date, game_time, location, age, gender, division, home_team, away_team)
VALUES (%s, %s, %s, %s, %s, %s, %s, %s)
"""


def parse_date(raw):
    return datetime.strptime(raw.strip(), "%m/%d/%Y").strftime("%Y-%m-%d")


def parse_time(raw):
    # Strip timezone suffix (EST, EDT) if present
    parts = raw.strip().split()
    return parts[0] if parts else None


def main():
    parser = argparse.ArgumentParser(description="Import soccer schedule CSV into MySQL")
    parser.add_argument("--host", default="localhost")
    parser.add_argument("--port", type=int, default=3306)
    parser.add_argument("--user", required=True)
    parser.add_argument("--password", default="")
    parser.add_argument("--database", default="soccer_schedule")
    args = parser.parse_args()

    conn = mysql.connector.connect(
        host=args.host,
        port=args.port,
        user=args.user,
        password=args.password,
        database=args.database,
    )
    cursor = conn.cursor()
    cursor.execute(CREATE_TABLE)

    inserted = 0
    skipped = 0

    with open(CSV_FILE, newline="", encoding="utf-8") as f:
        reader = csv.reader(f)
        next(reader)  # skip header row

        for row in reader:
            # Skip blank/separator rows
            if not row[0].strip():
                continue

            try:
                game_date = parse_date(row[0])
                game_time = parse_time(row[1])
                location  = row[2].strip() or None
                age       = row[3].strip() or None
                gender    = row[4].strip() or None
                division  = row[5].strip() or None
                home_team = row[6].strip() or None
                away_team = row[7].strip() or None
            except (ValueError, IndexError) as e:
                print(f"Skipping row {reader.line_num}: {e} — {row}")
                skipped += 1
                continue

            cursor.execute(INSERT_ROW, (game_date, game_time, location, age, gender, division, home_team, away_team))
            inserted += 1

    conn.commit()
    cursor.close()
    conn.close()
    print(f"Done: {inserted} rows inserted, {skipped} skipped.")


if __name__ == "__main__":
    main()
